### PR TITLE
TP-6521 add opt-in checkbox to user registration page

### PR DESCRIPTION
**Pull request by Tr4pSt3R**

### CODE REVIEW

File: app/views/devise/registrations/new.html.erb [hunk 1]
```diff
@@ -108,6 +108,14 @@
             </div>
           </fieldset>
 
+          <fieldset class="form__group">
+            <div class="form__group-item">
+              <%= f.label :opt_in, class: "form__label-heading" do %>
+                <%= f.check_box :opt_in, class: "form__group-input" %>
+                <%= t("activerecord.attributes.user.opt_in") %>
```
Comment (tomas-stefano) [inline app/views/devise/registrations/new.html.erb]: double quotes

File: app/views/devise/registrations/new.html.erb [hunk 2]
```diff
@@ -108,6 +108,14 @@
             </div>
           </fieldset>
 
+          <fieldset class="form__group">
+            <div class="form__group-item">
+              <%= f.label :opt_in, class: "form__label-heading" do %>
```
Comment (tomas-stefano) [inline app/views/devise/registrations/new.html.erb]: double quotes

**Comment by Tr4pSt3R** [comment]
@tomas-stefano. Your recommended fixes has been made. Please take another look

**Comment by Tr4pSt3R** [comment]
Hi @darokel or @asmega is it possible to have a look this please to see if that's sufficient for User-Customer mapping for CRM please? 
Thanks D

**Comment by darokel** [comment]
I think you'll also need to add your fields to `/lib/core/repository/customers/cream.rb` within the `FindMapper` `call` instance method.

**Comment by asmega** [comment]
@Tr4pSt3R I don't know. Can you not test it against the test box?

File: app/views/devise/registrations/new.html.erb [hunk 3]
```diff
@@ -13,12 +13,12 @@
     <p><%= t("authentication.mandatory_warning") %></p>
     <%= f.validation_summary %>
 
-    <div class="registration__row">
-      <div class="registration__field">
+    <div class='registration__row'>
```
Comment (Guntrisoft) [inline app/views/devise/registrations/new.html.erb]: The standard is double quotes for html attributes https://github.com/moneyadviceservice/frontend-code-standards#html-markup

File: app/views/devise/registrations/new.html.erb [hunk 4]
```diff
@@ -13,12 +13,12 @@
...
+      <div class='registration__field', id='first_name'>
```
Comment (asmega) [inline app/views/devise/registrations/new.html.erb]: no commas between attributes in HMTL

Comment (asmega) [inline app/views/devise/registrations/new.html.erb]: why did you add the `id`? what is it used for?

File: app/controllers/registrations_controller.rb [hunk 5]
```diff
@@ -24,6 +24,7 @@ def configure_permitted_parameters
     devise_parameter_sanitizer.for(:sign_up) << :first_name
     devise_parameter_sanitizer.for(:sign_up) << :post_code
     devise_parameter_sanitizer.for(:sign_up) << :newsletter_subscription
+    devise_parameter_sanitizer.for(:sign_up) << :opt_in
```
Comment (asmega) [inline app/controllers/registrations_controller.rb]: is `opt_in` the best name? what are we opting into?

File: app/views/devise/registrations/new.html.erb (hunk 4 above)
Comment (Tr4pSt3R) [inline app/views/devise/registrations/new.html.erb]: @asmega the idea was added before I realised that there was a gem called SitePrism which, was capable of picking up this element without having to explicitly set an id. 

I will remove this as well as fix the comma in the HTML attributes

**Comment by asmega** [comment]
:+1:

**Comment by darokel** [comment]
:+1:
